ios/brave-ios/Sources/BraveWallet/Crypto/CryptoView.swift

```diff
@@ -303,8 +334,36 @@ public struct CryptoView: View {
       })
     )
     .environment(\.webImageDownloader, webImageDownloader)
-    .onChange(of: visibleScreen) { newValue in
-      if case .panelUnlockOrSetup = presentingContext, newValue == .crypto {
+    .onChange(of: visibleScreen) { oldValue, newValue in
+      if newValue == .crypto {
+        if case .panelUnlockOrSetup = presentingContext {
+          // wallet is unlocked from wallet panel
+          dismissAction()
+        } else if case .default(_) = presentingContext {
+          // wallet is unlocked from menu
+          if FeatureList.kBraveWalletWebUIIOS?.enabled == true {
+            // need to open wallet webui
+            openWalletURLAction?(.brave.wallet.home)
+          }
+        } else if case .webUI(let action) = presentingContext {
+          if action == .unlock {
+            // wallet is unlocked from wallet webui
+            dismissAction()
+          } else if case .onboarding(_) = action {
+            // onboarding is completed from wallet webui
+            dismissAction()
+          }
+        }
+      }
```

Comment:
This may be better with a `switch` case now that there are multiple checks, e.g.

```swift
guard let newValue == .crypto else { return }
switch presentingContext {
case .panelUnlockOrSetup, .webUI(.unlock), .webUI(.onboarding):
  dismissAction()
case .default:
  ...
default:
  break
}
```